pkg/cmd/cli/cmd/rsync/rsyncargs.go

```diff
@@ -0,0 +1,196 @@
+
+	f.Bool("update", false, "")
+	f.Bool("inplace", false, "")
+	f.Bool("existing", false, "")
+	f.Bool("force", false, "")
+	f.BoolP("ignore-times", "I", false, "")
+	f.BoolP("prune-empty-dirs", "m", false, "")
+	f.BoolP("times", "t", false, "")
+	f.Bool("progress", false, "")
+	f.Bool("stats", false, "")
+	f.BoolP("itemize-changes", "i", false, "")
+	f.StringP("filter", "f", "", "")
+	f.String("exclude", "", "")
+	f.String("include", "", "")
+
+	allowedRsyncArgs = f
+
+}
+
+// Type returns argument type
+func (a *RsyncArgs) Type() string {
+	return "rsync-args"
+}
+
+// String returns single string with all arguments
+func (a *RsyncArgs) String() string {
+	return strings.Join(*a, " ")
+}
+
+// Set sets the argument list from a single string
+func (a *RsyncArgs) Set(value string) error {
+	// Parse the string into an array of arguments
+	args, err := parseArgs(value)
+	if err != nil {
+		return err
+	}
+	// Determine if the arguments are for valid flags
+	err = allowedRsyncArgs.Parse(args)
+	if err != nil {
+		return err
+	}
+	*a = args
+	return nil
+}
+
+// states for argument list parser
+const (
+	normal = iota
+	inArgument
+	inSingleQuote
+	inDoubleQuote
+	inEscape
+)
+
+// simple functions to improve readability of parser
+
+func isSingleQuote(c rune) bool {
+	return c == '\''
+}
+
+func isDoubleQuote(c rune) bool {
+	return c == '"'
+}
+
+func isEscape(c rune) bool {
+	return c == '\\'
+}
+
+func isSpace(c rune) bool {
+	return unicode.IsSpace(c)
+}
+
+// parseArgs uses a simple state machine to parse the arguments in a bash-style commmand line.
+// It will recognize single-quoted and double-quoted strings. Escaped characters are only
+// recognized in double-quoted strings
+// State machine states: normal (starting state), inArgument, inSingleQuote, inDoubleQuote, escapeChar
+// Transitions:
+// normal:
+//    space -> normal
+//    single-quote -> inSingleQuote
+//    double-quote -> inDoubleQuote
+//    escape -> inEscape (nextState = inArgument)
+//    other -> inArgument (accumulate char for arg)
+// inArgument:
+//    space -> normal (save and reset current arg)
+//    single-quote -> inSingleQuote
+//    double-quote -> inDoubleQuote
+//    escape  -> inEscape (nextState = inArgument)
+//    other -> inArgument (acccumulate char for arg)
+// inSingleQuote
+//    space -> inSingleQuote (accumulate char for arg)
+//    single-quote -> inArgument
+//    double-quote -> inSingleQuote (accumulate char for arg)
+//    escape-char -> inSingleQuote (accumulate char for arg)
+//    other -> inSingleQuote (accumulate char for arg)
+// inDoubleQuote
+//    space -> inDoubleQuote (accmulate char for arg)
+//    single-quote -> inDoubleQuote (accmulate char for arg)
+//    double-quote -> inArgument
+//    escape-char -> inEscape (nextState = inDoubleQuote)
+//    other -> inDoubleQuote (accmulate char for arg)
+// inEscape
+//    any -> nextState (accumulate char for arg)
+func parseArgs(str string) ([]string, error) {
```

Comment:
I found a couple of go libraries that did this (with caveats)
https://github.com/mattn/go-shellwords - no license
https://github.com/mgutz/str - panics on errors

I need to parse the different string args so I can pass them to the go cmd.exec ... alternatively, we could just white-list the args we accept and allow you to specify them directly on the command.